### PR TITLE
CaptureAll

### LINKED ISSUE #1243: empty path when using CaptureAll results to [""]
Expected would be `[]`.

### PR DESCRIPTION
**Pull request by guygastineau**
closes #1243 

In the end it was a simple fix, but new tests were necessary.  I also modified the existing tests for the original test API for CaptureAll.  Using the sum of a list let the bug from #1243 linger.  Now the tests will catch any regressions.  I am not sure if there.

I tried combining the "rooted" capture all API at the end of the legs and arms API, but then it caught the parsing failure tests for the leg route, so I figured it might as well be another router.

I tried to be tidy.

### CODE REVIEW

**Comment by guygastineau** [comment]
@domenkozar please let us know if this solves your issue.

**Comment by guygastineau** [comment]
This PR does change the data processed by handlers for CaptureAll apis.  This might mean some kind of breaking change.  The difference is:
```
// was ["", ""], but now it is [""]
/some-route/ was [""], but now it is []
/some-route// was ["", ""], but now it is [""]
```

**Comment by guygastineau** [comment]
I forgot to make a file in changeling.d. sorry

**Comment by alpmestan** [comment]
FWIW, The new behaviour seems more correct to me.

**Comment by guygastineau** [comment]
@alpmestan great.  The side by side string and int tests might be redundant.  Or maybe the community wants new tests?  I think I got the tests we need to understand my change and prevent regression, but I'd love if someone tried to poke holes in that confidence.

File: changelog.d/CaptureAll [hunk 1]
```diff
@@ -0,0 +1,14 @@
+synopsis: Bugfix - CaptureAll produces [""] for empty paths due to trailing slash.
+prs: #1516
+issues: #1243
+
+description: {
+
+CaptrueAll resulted in `[""]` for empty paths due to trailing slash.  Similar
```
Comment (alpmestan) [inline changelog.d/CaptureAll:7]: `CaptureAll` *

**Review by alpmestan — APPROVED** [review]
Just a typo, otherwise LGTM.

File: changelog.d/CaptureAll (hunk 1 above)
Comment (guygastineau) [reply changelog.d/CaptureAll:7]: Thanks you.  I'll get the typo fixed this afternoon EST.

Comment (guygastineau) [reply changelog.d/CaptureAll:7]: Oh, I forgot to get to this.  I'll get it today.

Comment (guygastineau) [reply changelog.d/CaptureAll]: I amended and force pushed.  Apologies to anyone who might have fetched my branch.

**Comment by alpmestan** [comment]
CI jobs running now. If it goes through, this is IMO good to go. @akhesaCaro @gdeest @tchoutri Your opinion?

**Review by tchoutri — APPROVED** [review]
I'm overall satisfied with this PR, thank you very much for the feature. :)

**Comment by guygastineau** [comment]
Is there a reason why this hasn't been merged yet other than everyone being busy and letting slip through the cracks?

**Pull request merged**